skywalking/client/http.py

```diff
@@ -109,3 +110,20 @@ def report(self, generator):
                 } for span in segment.spans]
             })
             logger.debug('report traces response: %s', res)
+
+
+class HttpLogDataReportService(TraceSegmentReportService):
+    def __init__(self):
+        proto = 'https://' if config.force_tls else 'http://'
+        self.url_report = proto + config.collector_address.rstrip('/') + '/v3/logs'
+        self.session = requests.Session()
+
+    def fork_after_in_child(self):
+        self.session.close()
+        self.session = requests.Session()
+
+    def report(self, generator):
+        for log_data in generator:
+            json_string = json_format.MessageToJson(log_data)
+            res = self.session.post(self.url_report, json=[json.loads(json_string)])
```

Comment:
Actually, looking at this suggests that the `/v3/logs` endpoint can take an array of logs so a batch send could be done. Change to:
```py
def report(self, generator):
    json = [json_format.MessageToJson(log_data) for log_data in generator]
    res = self.session.post(self.url_report, json=json)
```
Your call if you want to do now.

Reply:
Umm, should I make a new config entry allowing the user to choose whether batch or not?

Reply:
No, batch is the right way to go (assuming this endpoint does take arrays, which is what you should check).

Reply:
Got it, it does take arrays.